### DISCUSSION TITLE
Multiple Input+Output directories using feature-to-layer approach

### DISCUSSION THREAD

**Discussion by hgiesel** [slang-i18n/slang#123] [Q&A] [answered]
Can I have a project structure like this:

```
lib
    feature1
        models
        controllers
        i18n
            foo.i18n.json
    feature2
        models
        controllers
        i18n
            bar.i18n.json
```

And use slang to generate its output files into each `i18n` directory per feature?

**Comment by Tienisto** [comment, marked as answer]
Hello, this requires a lot of refactoring effort because "namespaces" are erased after being parsed. The internal model don't have knowledge about namespaces.

So generating per-locale is possible but per-namespaces is unfortunately not.

**Reply by hgiesel** [reply]
Ah, that's unfortunate, it would be a great feature to have, but I understand if there are other priorities :)
